### ISSUE TITLE
Passthrough video metadata isn't working properly

### ISSUE THREAD

**Issue by mpek** [RandomEngy/VidCoder#1255]
### Problem Description

MKV h.264 to MP4 HEVC conversion.

Tags are written to different tagfields or get discarded in output files: 

DIRECTOR > ALBUMARTIST
COMMENT is empty in new file
YEAR is empty in new file
LANGUAGE is empty in new file
SUBTITLE is empty in new file
MEDIATYPE is empty in new file

All tags above are filled in source file. 
Please let me know if you need an encode log.

### What version of VidCoder are you running?

10.9 Beta

**Comment by RandomEngy** [comment]
Can you post a scan log from the source you're using? It should show what metadata is being picked up by the HandBrake library.

**Comment by mpek** [comment]
I assume that you don't need the whole scan log and this would be enough, right?
 
VC [19:10:24] VidCoder 11.1 Beta
VC [19:10:24] Starting scan: D:\Test with Tags.mkv
HB [19:10:24] CPU: AMD Ryzen 9 5900X 12-Core Processor
HB [19:10:24]  - logical processor count: 24
HB [19:10:24] Intel Quick Sync Video support: no
HB [19:10:24] hb_scan: path=D:\Test with Tags.mkv, title_index=0
HB Input #0, matroska,webm, from 'D:\Test with Tags.mkv':
HB   Metadata:
HB     encoder         : libmakemkv v1.17.7 (1.3.10/1.5.2) win(x64-release)
HB     creation_time   : 2024-06-19T12:17:39.000000Z
HB     ARTIST          : Tom Cruise/Colin Farrell/Max von Sydow/Samantha Morton/Kathryn Morris/Steve Harris/Neal McDonough/Lois Smith/Peter Stormare
HB     COMMENT         : Im Jahr 2054 hat es seit Jahren keinen Mord mehr gegeben, denn das Department of Precrime sorgt dafÃ¼r, dass es gar nicht mehr erst dazu kommen kann. Aufgrund einer prÃ¤kognitiven Ãœberwachungstechnik sind die Cops in der Lage, zukÃ¼nftig zu begehende Ver
HB     DIRECTOR        : Steven Director
HB     GENRE           : Sci-Fi/Thriller/Action
HB     LANGUAGE        : deu/eng
HB     ORIGINAL_MEDIA_TYPE: FHD
HB     SUBTITLE        : deu/eng
HB     TITLE           : Minority Document 2002 deu-eng FHD
HB     DATE_RECORDED   : 2002

**Comment by mpek** [comment]
Hope this helps and you don't need the whole log: 

VC [19:10:24] VidCoder 11.1 Beta
VC [19:10:24] Starting scan: D:\Test with Tags.mkv
HB [19:10:24] CPU: AMD Ryzen 9 5900X 12-Core Processor
HB [19:10:24]  - logical processor count: 24
HB [19:10:24] Intel Quick Sync Video support: no
HB [19:10:24] hb_scan: path=D:\Test with Tags.mkv, title_index=0
HB Input #0, matroska,webm, from 'D:\Test with Tags.mkv':
HB   Metadata:
HB     encoder         : libmakemkv v1.17.7 (1.3.10/1.5.2) win(x64-release)
HB     creation_time   : 2024-06-19T12:17:39.000000Z
HB     ARTIST          : Tom Cuise/Colin Fallell/Max von Sydo/Samantha Moton/Kathy Moris/Stevie Haris/Neal McDono/Lois Lane/Peter Storm
HB     COMMENT         : Im Jahr 2054 hat es seit Jahren keine Süßigkeiten mehr gegeben, denn das Department of Sweets sorgt dafÃ¼r, dass es gar nicht mehr dazu kommen kann. Aufgrund einer prÃ¤kognitiven Ãœberwachungstechnik sind die Mitarbeiter in der Lage, zukÃ¼nftig zu begehende Süß
HB     DIRECTOR        : Steven Playhill
HB     GENRE           : Sci-Fi/Thriller/Action
HB     LANGUAGE        : deu/eng
HB     ORIGINAL_MEDIA_TYPE: FHD
HB     SUBTITLE        : deu/eng
HB     TITLE           : Minority Document 2002 deu-eng FHD
HB     DATE_RECORDED   : 2002

**Comment by RandomEngy** [comment]
Do you have this option enabled in the Picker window?
![image](https://github.com/RandomEngy/VidCoder/assets/9450861/fae290a2-b82c-4400-9b6b-ddde8b743840)

If so, and you're still seeing the issue, please post a full encode log.

**Comment by mpek** [comment]
Yes! Then I made the encodings and recognised that the tags are not taken over to the new file how they are being expected.

**Comment by RandomEngy** [comment]
Please post a full scan and encode log, so I have a better idea of what is going on.

**Comment by mpek** [comment]
I made a 220MB MKV with all issued tag fields and let VidCoder Beta 11.3 generate a MP4 (116MB ).
If you'd need the file(s), please let me know. 
[2024-07-07 13.37.43 Scan Different Audiotracks Different Subtitles 1min.txt](https://github.com/user-attachments/files/16119004/2024-07-07.13.37.43.Scan.Different.Audiotracks.Different.Subtitles.1min.txt)
[2024-07-07 13.38.03 Encode Different Audiotracks Different Subtitles 1min.mp4-succeeded.txt](https://github.com/user-attachments/files/16119005/2024-07-07.13.38.03.Encode.Different.Audiotracks.Different.Subtitles.1min.mp4-succeeded.txt)

**Comment by RandomEngy** [comment]
Thanks. It looks like the metadata was picked up by the scan, and VidCoder passed through the metadata in the encode job JSON. But you say the metadata is not present in the output file? Having the source video may help: the next step would be to see if the HandBrake UI also has the same behavior. It may be an issue with the underlying HandBrake core.

**Comment by mpek** [comment]
Okay, where can I upload the source file for you?

**Comment by RandomEngy** [comment]
Google Drive? https://transfer.pcloud.com/ has also worked in the past for me.

**Comment by mpek** [comment]
Should be available here: 

https://transfer.pcloud.com/download.html?code=5ZruWT0ZzIbP0Sdlh1fZPKt77ZYyQvcbzgSrpXqrHfEvQHjzn9CpP7

**Comment by RandomEngy** [comment]
Sorry, I was on vacation and that download link expired. :(

**Comment by mpek** [comment]
Hope you had a nice vacation! 
Try this link please: 
https://transfer.pcloud.com/download.html?code=5ZlhmP0Zh9SkGaUHEEXZlLHk7ZOc5FqyDT7G8ht0dsJbIuOfvMz4W7

**Comment by RandomEngy** [comment]
Alright, I've been testing and when I encode to an MP4 container I do see them get passed through in MediaInfo. It doesn't work when I output to an MKV container.

Choosing x265 as the video encoder didn't seem to break this. The next thing for me to try would be to use your encoding preset. Can you export your preset and share the .vjpreset file?

**Comment by mpek** [comment]
I just tried MediaInfo, but it's not showing all possible tags with current release. 

You could try to compare the files in MKVToolNix GUI and see all included tags. 
On the left side, right below the Multiplexer you can change to the Info-tool, where you can drop the source mkv and the encoded mkv, right next to it. Then scroll down to "tags" where you can open by clicking on the little triangle to reveal the information. All tags included are shown. 

I use mp3tag to edit tags with mp4 and mkv, and music files for sure, as it was an mp3 tagger when it came out. The author included every tag possible, even for video files. To comfortably edit the tags, I configured the fields to my liking. By default, it does not show all tag fields possible for video files. If you want to try it, i can send you my config file to get you started quickly.

**Comment by RandomEngy** [comment]
That preset outputs to an MKV container, which does not support the metadata tags. That's an issue with the HandBrake API that VidCoder is using, so you'd need to ask the HandBrake team about metadata passthrough for MKV.

**Comment by mpek** [comment]
Well, MKV does support metadata tags, but if there is an issue with Handbrake not supporting MKV fully that's a pitty... 

The same preset is being used to output to MP4 and it shows the same issue described initially. 
Does this format has the same problem you described for MKV?

**Comment by RandomEngy** [comment]
If I take your preset and change it to MP4 it seems to work just fine for me: the tags are passed through and show in MediaInfo.

**Comment by mpek** [comment]
Yeah, some of them, and some on different tag fields and some are being deleted, it is as described initially. MediaInfo is not as accurate as it's being needed for this topic, options described [here](https://github.com/RandomEngy/VidCoder/issues/1255#issuecomment-2250191776).

**Comment by RandomEngy** [comment]
Can you be more specific about what metadata is missing? I saw all of the metadata show up that was recognized by the HandBrake API scan. If it doesn't show up in the encode log, that means the HandBrake API does not recognize it.

**Comment by mpek** [comment]
I'll try to show it to you in a hopefully obvious way. Here are some screenshot of the tag tool that i use of before and after the encoding(s): 

Before encoding: 
![Before encoding Mp3tag](https://github.com/user-attachments/assets/3f9323ee-9192-43a0-8fe5-d063f59c34d9)

After encoding to MP4:
![After encoding to MP4](https://github.com/user-attachments/assets/8b2823cb-2b81-475b-95ee-a5952d8d0d83)

After encoding to MKV:
![After encoding to MKV](https://github.com/user-attachments/assets/cd30ff7b-25f9-41f6-9d3b-cc5b0f399ba9)

I made a new package with all the files used. Here's the link: https://transfer.pcloud.com/download.html?code=5ZGKHO0ZatLXUmObGSFZCVRV7ZioWqkmEJR5ki6JTR4QEPT5mhtIBy
